### PR TITLE
Fixes DEVTOOLING-851 and DEVTOOLING-848 for genesyscloud_telephony_providers_edges_site_outbound_route

### PR DESCRIPTION
**Pull request by bbbco**
Fixes:
- DEVTOOLING-851 - genesyscloud_telephony_providers_edges_site_outbound_routes resource should be handling individual routes
- DEVTOOLING-848 - Adds a data resource for genesyscloud_telephony_providers_edges_site_outbound_route and adds unit tests

### CODE REVIEW

File: genesyscloud/telephony_providers_edges_site_outbound_route/data_source_genesyscloud_telephony_providers_edges_site_outbound_route.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceSiteOutboundRouteRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	if exists := featureToggles.OutboundRoutesToggleExists(); !exists {
```
Comment (carnellj-genesys) [inline genesyscloud/telephony_providers_edges_site_outbound_route/data_source_genesyscloud_telephony_providers_edges_site_outbound_route.go:19]: Hey Brian is this a brand new data source.  If it then I don't think we need a feature toggle here.  Just  curious as to why you put one here.

File: genesyscloud/telephony_providers_edges_site_outbound_route/data_source_genesyscloud_telephony_providers_edges_site_outbound_route.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		siteId, retryable, resp, err := sp.siteProxy.GetSiteIdByName(ctx, name)
+		if err != nil {
+			if retryable {
+				return retry.RetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("failed to get site %s", name), resp))
+			}
+
+			return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("error requesting site %s | error: %s", name, err), resp))
+		}
+
+		d.SetId(siteId)
```
Comment (charliecon) [inline genesyscloud/telephony_providers_edges_site_outbound_route/data_source_genesyscloud_telephony_providers_edges_site_outbound_route.go:38]: it looks like this just functions the same as the sites data source? Shouldn't we be using the name or ID of the site along with the outbound route name to find the ID of the outbound route?

File: genesyscloud/telephony_providers_edges_site_outbound_route/resource_genesyscloud_telephony_providers_edges_site_outbound_route_schema.go [hunk 3]
```diff
@@ -109,3 +110,18 @@ func SiteExporterOutboundRoute() *resourceExporter.ResourceExporter {
 		},
 	}
 }
+
+// DataSourceSite registers the genesyscloud_telephony_providers_edges_site_outbound_route data source
+func DataSourceSiteOutboundRoute() *schema.Resource {
+	return &schema.Resource{
+		Description: "Data source for Genesys Cloud Site Outbound Routes. Select by a using the name of the associated site name",
```
[the remark this answers is not in the document]
Comment (bbbco) [reply genesyscloud/telephony_providers_edges_site_outbound_route/resource_genesyscloud_telephony_providers_edges_site_outbound_route_schema.go:117]: Thank you

File: genesyscloud/telephony_providers_edges_site_outbound_route/data_source_genesyscloud_telephony_providers_edges_site_outbound_route.go (hunk 1 above)
Comment (bbbco) [reply genesyscloud/telephony_providers_edges_site_outbound_route/data_source_genesyscloud_telephony_providers_edges_site_outbound_route.go:19]: I included the matching feature toggle to the existing resource.

**Comment by bbbco** [comment]
I'm going to update this PR with some fixes required for restructuring the `genesyscloud_telephony_providers_edges_site_outbound_route` resource as discussed earlier with @dginty4 .

**Comment by bbbco** [comment]
@charliecon @carnellj-genesys @dginty4 this PR is ready for review with all tests passing

**Review by carnellj-genesys — APPROVED** [review]

File: genesyscloud/telephony_providers_edges_site_outbound_route/genesyscloud_telephony_providers_edges_site_outbound_route_proxy.go [hunk 4]
```diff
@@ -15,128 +16,202 @@ out during testing.
...
-func updateSiteOutboundRouteFn(ctx context.Context, p *siteOutboundRoutesProxy, siteId string, outboundRouteId string, outboundRoute *platformclientv2.Outboundroutebase) (*platformclientv2.Outboundroutebase, *platformclientv2.APIResponse, error) {
+func updateSiteOutboundRouteFn(ctx context.Context, p *siteOutboundRouteProxy, siteId string, outboundRouteId string, outboundRoute *platformclientv2.Outboundroutebase) (*platformclientv2.Outboundroutebase, *platformclientv2.APIResponse, error) {
 	return p.edgesApi.PutTelephonyProvidersEdgesSiteOutboundroute(siteId, outboundRouteId, *outboundRoute)
 }
 
 // deleteSiteOutboundRouteFn is an implementation function for deleting an outbound route for a Genesys Cloud Site
-func deleteSiteOutboundRouteFn(ctx context.Context, p *siteOutboundRoutesProxy, siteId string, outboundRouteId string) (*platformclientv2.APIResponse, error) {
+func deleteSiteOutboundRouteFn(ctx context.Context, p *siteOutboundRouteProxy, siteId string, outboundRouteId string) (*platformclientv2.APIResponse, error) {
 	return p.edgesApi.DeleteTelephonyProvidersEdgesSiteOutboundroute(siteId, outboundRouteId)
 }
```
[the remark this answers is not in the document]
Comment (charliecon) [reply genesyscloud/telephony_providers_edges_site_outbound_route/genesyscloud_telephony_providers_edges_site_outbound_route_proxy.go:217]: This is really only a problem that pops up in our test cases.

**Review by charliecon — APPROVED** [review]

**Review by dginty4 — APPROVED** [review]

File: genesyscloud/telephony_providers_edges_site_outbound_route/genesyscloud_telephony_providers_edges_site_outbound_route_proxy.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (bbbco) [reply genesyscloud/telephony_providers_edges_site_outbound_route/genesyscloud_telephony_providers_edges_site_outbound_route_proxy.go:223]: Thanks for catching this! It's updated in the PR.

**Review by HemanthDogiparthi12 — APPROVED** [review]

**Pull request merged**